src/main/java/com/xebialabs/overthere/gcp/GcpMetadataKeyManager.java

```diff
@@ -211,11 +215,19 @@ private void checkForOperationErrors(final Operation operation) {
     }
 
     private Compute createComputeService() {
-        return new Compute.Builder(
-                httpTransport,
-                jsonFactory,
-                new HttpCredentialsAdapter(projectCredentials.getCredentials())
-        ).setApplicationName(applicationName).build();
+        if(projectCredentials.getCredential() instanceof Credential) {
```

Comment:
I would better here use null check against projectCredentials.getCredential(), instead of instanceof.

Reply:
Sure I will change accordingly :)